### ISSUE TITLE
V parser segfault, when run with `-silent`, discovered through radamsa

### ISSUE THREAD

**Issue by 4hnme** [vlang/v#22085]
### Describe the bug

in silent mode, compiler throws a segfault when encountering 100 nested scopes

### Reproduction Steps

download [source code](https://pastebin.com/UPnQSsrz) and run `v -silent x.v`

### Expected Behavior

it shouldn't segfault

### Current Behavior

it segfaults

### V version

0.4.7 39e550f.5e8bb21

### Environment details (OS name and version, etc.)

V full version: V 0.4.7 39e550f.5e8bb21
OS: linux, "Manjaro Linux"
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i5-2500 CPU @ 3.30GHz

vexe: /home/hotsadboi/Documents/v/v
vexe mtime: 2024-08-20 17:11:34

vroot: OK, value: /home/hotsadboi/Documents/v
VMODULES: OK, value: /home/hotsadboi/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.46.0
Git vroot status: weekly.2024.34-37-g5e8bb21d
.git/config present: true

CC version: cc (GCC) 14.1.1 20240720
thirdparty/tcc status: thirdparty-linux-amd64 40e5cbb5

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

**Comment by spytheman** [comment]
Still segfaults on latest V 18f89991efbeea490621a2a961d3b926f6592e8f .

**Comment by spytheman** [comment]
Fixed on latest master, after e3ceb5881a793c0924a4e3f03e6e2238906e7731 :
```
#1 08:34:28 ^ master /v/oo>./v -g self
V self compiling (-g -o v2)...
V built successfully as executable "v".
#0 08:34:34 ^ master /v/oo>./v -g cmd/tools/measure/parser_speed.v
#0 08:34:40 ^ master /v/oo>cmd/tools/measure/parser_speed  /v/misc/2024_08_22__21/issue_22085.v
[cmd/tools/measure/parser_speed.v:19] main.comments_mode: skip_comments
     24023us     487460     487517          1             1        103   /v/misc/2024_08_22__21/issue_22085.v
---------------------------------------------------------------------------------------------------------------------------------------------------
        Time     Tokens      Bytes      Lines   Bytes/Token     Errors
---------------------------------------------------------------------------------------------------------------------------------------------------
     24023us     487460     487517          1             1        103   Parser speed: 20.294 MB/s,       41.6 lines/s,   1 thread(s),     1 files.
#0 08:34:48 ^ master /v/oo>./v -silent /v/misc/2024_08_22__21/issue_22085.v
#1 08:34:55 ^ master /v/oo>
```
